pkg/infinity/client.go

```diff
@@ -200,7 +200,7 @@ func (client *Client) GetResults(ctx context.Context, query models.Query, reques
 		return string(bodyBytes), http.StatusOK, 0, nil
 	}
 	switch strings.ToUpper(query.URLOptions.Method) {
-	case http.MethodPost:
+	case http.MethodPost, http.MethodPatch, http.MethodPut, http.MethodDelete:
```

Comment:
nit pick - this but this could be actually flipped - have `case http.MethodGet` as we could use the default handling for everything besides get method.